crates/burn-tensor/src/tensor/api/int.rs

```diff
@@ -70,6 +71,57 @@ where
         Tensor::new(B::int_into_float(self.primitive))
     }
 
+    /// Produces an indices tensor for the given shape and device.
+    /// The resulting tensor contains coordinates corresponding to each element in the shape at dimension D.
+    ///
+    /// # Arguments
+    ///
+    /// * `shape` - The shape specifying the dimensions of the tensor.
+    /// * `device` - The device to create the tensor on.
+    ///
+    /// # Panics
+    ///
+    /// Panics if `D2` is not equal to `D+1`.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    ///    use burn_tensor::Int;
+    ///    use burn_tensor::{backend::Backend, Shape, Tensor};
+    ///    fn example<B: Backend>() {
+    ///        let device = Default::default();
+    ///        let result = Tensor::<B, 2, Int>::indices::<3>(Shape { dims: [2, 3] }, &device);
+    ///        println!("{}", result);
+    ///    }
+    /// ```
+    pub fn indices<const D2: usize>(shape: Shape<D>, device: &B::Device) -> Tensor<B, D2, Int> {
```

Comment:
Usually indices will be produced to use with a matrix/tensor (i.e., indexing in `i, j` like [`np.indices`](https://numpy.org/doc/stable/reference/generated/numpy.indices.html)) but I think the intention with this method is to produce a grid in cartesian space. With that in mind, I think I would make it a bit more explicit in the method's doc and rename the method to something like `nd_grid`, `meshgrid` or even more explicit `cartesian_grid` (also opened to suggestions).

Reply:
`cartesian_grid` seems to be a good name!